src/Generating/Trees.cpp

```diff
@@ -58,7 +59,8 @@ static const Vector3d & pickBranchDirection(const cNoise a_Noise, Vector3i a_Blo
 		}
 	};
 
-	size_t index = static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockPos.x, a_BlockPos.y + a_Seq, a_BlockPos.z)) % directions.size();
+	// Old: size_t index = static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockPos.x, a_BlockPos.y + a_Seq, a_BlockPos.z)) % directions.size();
+	size_t index = static_cast<size_t>(GetRandomProvider().RandInt(directions.size() - 1));
```

Comment:
You can't use randomness is the terrain generator, if you do then world seeds won't reproduce.

Reply:
Fair Point. Back to the drawing board

Reply:
changed with new comit